pkg/controller/controller_test.go

```diff
@@ -228,14 +249,15 @@ func waitForStatefulSets(
 				seen++
 			}
 		}
+		if len(updatedPods) > 0 {
+			podUpdateGroups = append(podUpdateGroups, updatedPods)
+			updatedPods = nil
+		}
 		mu.Unlock()
 
 		if seen != len(opts.expectedStatefulSets) {
 			time.Sleep(100 * time.Millisecond)
-			continue
```

Comment:
This continue and break logic actually caused the test method to exit before statefulset updates were finished so just remove and rely on the iterations limit.